File: Site/mariyacompany.web/src/app/models/employee.ts
import { CompanyPositions } from "./company.positions";

export class Employee {
  constructor(
    public id?: number,
    public firstName?: string,
    public middleName?: string,
    public lastName?: string,
    public birthday?: Date,
    public employmentDate?: Date,
    public departmentId?: number,
    public companyPositionId?: number,
    public summary?: string,
    public departmentString?: string,
    public companyPositionString?: string,
    public birthdayString?: string,
    public employmentDateString?: string,
    public companyPosition?: CompanyPositions) { }
}

File: MariyaCompany.Application/Abstractions/Entity/Employee.cs
using System;

namespace MariyaCompany.Application.Abstractions.Entity
{
    public class Employee : DpEntity
    {
        /// <summary>
        /// имя
        /// </summary>
        public string FirstName { get; set; }

        /// <summary>
        /// отчество
        /// </summary>
        public string MiddleName { get; set; }

        /// <summary>
        /// фамилия
        /// </summary>
        public string LastName { get; set; }

        public DateTime Birthday { get; set; }

        public DateTime EmploymentDate { get; set; }

        public int DepartmentId { get; set; }

        public int CompanyPositionId { get; set; }

        public virtual Department Department { get; set; }

        public virtual CompanyPosition CompanyPosition { get; set; }
    }
}
